File: Backend/src/main/java/lan/dk/podcastserver/service/health/UpdaterHealthIndicator.java
package lan.dk.podcastserver.service.health;

import lan.dk.podcastserver.business.update.UpdatePodcastBusiness;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.actuate.health.AbstractHealthIndicator;
import org.springframework.boot.actuate.health.Health;
import org.springframework.stereotype.Component;

import static java.util.Objects.isNull;

/**
 * Created by kevin on 18/07/2016.
 */
@Component
@RequiredArgsConstructor
public class UpdaterHealthIndicator extends AbstractHealthIndicator {

    private final UpdatePodcastBusiness updater;

    @Override
    protected void doHealthCheck(Health.Builder builder) throws Exception {
        // @formatter:off
        builder.up()
            .withDetail("lastFullUpdate", isNull(updater.getLastFullUpdate()) ? "none" : updater.getLastFullUpdate())
            .withDetail("isUpdating", updater.isUpdating())
            .withDetail("activeThread", updater.getUpdaterActiveCount())
        .build();
        // @formatter:on
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/service/health/UpdaterHealthIndicator.kt
package com.github.davinkevin.podcastserver.service.health

import com.github.davinkevin.podcastserver.business.update.UpdatePodcastBusiness
import org.springframework.boot.actuate.health.AbstractHealthIndicator
import org.springframework.boot.actuate.health.Health
import org.springframework.stereotype.Component

/**
 * Created by kevin on 18/07/2016.
 */
@Component
class UpdaterHealthIndicator(val updater: UpdatePodcastBusiness) : AbstractHealthIndicator() {

    @Throws(Exception::class)
    override fun doHealthCheck(builder: Health.Builder) {
        // @formatter:off
        builder.up()
            .withDetail("lastFullUpdate", updater.lastFullUpdate ?: "none")
            .withDetail("isUpdating", updater.isUpdating)
            .withDetail("activeThread", updater.updaterActiveCount)
        .build()
         // @formatter:on
    }
}
